src/test/java/com/jcabi/log/SupplierLoggerTest.java

```diff
@@ -0,0 +1,277 @@
+     */
+    @Test
+    public void warnIsEnabled() throws Exception {
+        final String text = "test6";
+        final org.apache.log4j.Logger logger = org.apache.log4j.Logger
+            .getLogger(this.getClass().getSimpleName());
+        logger.setLevel(Level.WARN);
+        Logger.withSupplier().warn(
+            this.getClass().getSimpleName(), "Warn enabled: %s",
+            new Supplier<String>() {
+                public String get() {
+                    return text;
+                }
+            }
+        );
+        final UnitTestAppender appender = (UnitTestAppender) logger
+            .getAppender(SupplierLoggerTest.APPENDER);
+        MatcherAssert.assertThat(
+            new String(appender.output().toByteArray()),
+            Matchers.containsString(text)
+        );
+    }
+
+    /**
+     * SupplierLogger does not log a message with INFO level if debug is not
+     * enabled.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void infoIsDisabled() throws Exception {
+        final org.apache.log4j.Logger logger = org.apache.log4j.Logger
+            .getLogger(this.getClass().getSimpleName());
+        logger.setLevel(Level.OFF);
+        Logger.withSupplier().info(
+            this.getClass().getSimpleName(), "Info disabled: %s",
+            new Supplier<String>() {
+                public String get() {
+                    return "test7";
+                }
+            }
+        );
+        final UnitTestAppender app = (UnitTestAppender) logger
+            .getAppender(SupplierLoggerTest.APPENDER);
+        MatcherAssert.assertThat(
+            new String(app.output().toByteArray()),
+            Matchers.isEmptyString()
+        );
+    }
+
+    /**
+     * SupplierLogger logs a message with INFO level if debug is enabled.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void infoIsEnabled() throws Exception {
+        final String text = "text8";
+        final org.apache.log4j.Logger logger = org.apache.log4j.Logger
+            .getLogger(this.getClass().getSimpleName());
+        logger.setLevel(Level.INFO);
+        Logger.withSupplier().info(
+            this.getClass().getSimpleName(), "Info enabled: %s",
+            new Supplier<String>() {
+                public String get() {
+                    return text;
+                }
+            }
+        );
+        final UnitTestAppender appender = (UnitTestAppender) logger
+            .getAppender(SupplierLoggerTest.APPENDER);
+        MatcherAssert.assertThat(
+            new String(appender.output().toByteArray()),
+            Matchers.containsString(text)
+        );
+    }
+
+    /**
+     * Cleans the appender's OutputStream after each test.
+     * This method is guaranteed to run even if an exception is thrown during
+     * a test run.
+     * @throws IOException If something goes wrong
+     */
+    @After
+    public void cleanLogs() throws IOException {
```

Comment:
@mkordas  This method clears the Appender's outputstream after each test execution, so the next test can have a fresh start. This wouldn't be a problem for the tests that write someting (since it's a contains() assertion), but it's a problem for those that are not supposed to log anything, which the assertion is that it should be empty.